### ISSUE TITLE
Records file changes between adjacent builds even there are no changes to codes.

### ISSUE THREAD

**Issue by BurningLutz** [webpack/webpack#5634]
**Do you want to request a *feature* or report a *bug*?**
report a bug.

**What is the current behavior?**
Records file changes a bit between builds even there are no changes to codes.
**If the current behavior is a bug, please provide the steps to reproduce.**
Here are my webpack config and codes:
```js
// webpack.config.js
'use strict'

const path = require('path')
const webpack = require('webpack')
const HtmlWebpackPlugin = require('html-webpack-plugin')

module.exports = {
  entry: {
    main: './src/index.js',
    vendor: [
      'react',
      'react-dom',
    ]
  },
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: '[name].[chunkhash:8].js',
    chunkFilename: '[name].[chunkhash:8].js'
  },
  resolve: {
    extensions: ['.js', '.jsx'],
    modules: ['./src', 'node_modules']
  },
  module: {
    rules: [{
      test: /\.js(x?)$/,
      exclude: /node_modules/,
      use: ['babel-loader']
    }]
  },
  plugins: [
    new HtmlWebpackPlugin({
      template: './public/index.html',
      inject: 'body'
    }),
    new webpack.optimize.UglifyJsPlugin({
      comments: false
    }),
    new webpack.optimize.MinChunkSizePlugin({
      minChunkSize: 10240
    }),
    new webpack.optimize.AggressiveMergingPlugin(),
    new webpack.optimize.CommonsChunkPlugin({
      name: ['vendor', 'manifest'],
      minChunks: Infinity
    }),
    new webpack.DefinePlugin({
      'process.env.NODE_ENV': JSON.stringify('production')
    }),
  ],
  recordsPath: path.resolve(__dirname, 'records.json')
}
```
```js
// index.js
import React from 'react'
import { render } from 'react-dom'

render(
  <div>hello world</div>,
  document.getElementById('root')
)
```
I built with config and codes above for the first time, the second time, and I got different records.json file.
The diff between two builds is something like:
```diff
--- a/records.json
+++ b/records.json
@@ -19,7 +19,7 @@
         "byName": {},
         "byBlocks": {},
         "usedIds": {
-          "0": 0
+          "1": 1
         }
       }
     }
```

**What is the expected behavior?**
records file should stay unchanged when there are no changes to codes or webpack config.
**If this is a feature request, what is motivation or use case for changing the behavior?**

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
Node: 8.1.3
webpack: 3.5.5
babel-loader: 7.1.1
babel-core: 6.26.0
OS: macOS Sierra 10.12.6

**Comment by sokra** [comment]
Thanks for your report. It would be great if you reduce your issue to a small reproducible example. Best put this example into a github repository together with instructions how to get to the problem.

**Comment by BurningLutz** [comment]
@sokra Thanks for the quick reply!
I made a minimized example in this [gist](https://gist.github.com/BurningLutz/048cc5efd75bf1c985940a50070a7cab) to reproduce the issue.
The `new HtmlWebpackPlugin()` in `webpack.config.js` seems important, the issue won't reproduce without it.

**Comment by JoeBernardi** [comment]
I encountered the same issue, except in my case the culprit seems to be extract-text-webpack-plugin.

From what I can tell, this is a result of both extract-text-webpack-plugin and html-weback-plugin triggering child compilations that don't emit any chunks, which seems to confuse the logic used to apply chunk IDs ([here](https://github.com/webpack/webpack/blob/master/lib/Compilation.js#L1157-L1202)).

I was able to get around this by [modifying](https://gist.github.com/JoeBernardi/fbee49e4cda9d656867ba8d0701cecc5) `RecordIdsPlugin` to only assign IDs to chunks that also have names, but it's possible a sturdier solution could be applied elsewhere.
